interface/patient_file/deleter.php

```diff
@@ -461,37 +461,53 @@ function popup_close() {
-                    } elseif ($payment) {
-                        echo xlt('payment') . " " . text($payment);
-                    } elseif ($billing) {
-                        echo xlt('invoice') . " " . text($billing);
-                    } elseif ($transaction) {
-                        echo xlt('transaction') . " " . text($transaction);
-                    }
-                    ?>
-                </span>
-                <?php echo xlt('and all subordinate data? This action will be logged'); ?>!
+        <form method='post' name="deletefrm" action='deleter.php?patient=<?php echo attr_url($patient) ?>&encounterid=<?php echo attr_url($encounterid) ?>&formid=<?php echo attr_url($formid) ?>&issue=<?php echo attr_url($issue) ?>&document=<?php echo attr_url($document) ?>&payment=<?php echo attr_url($payment) ?>&billing=<?php echo attr_url($billing) ?>&transaction=<?php echo attr_url($transaction); ?>&csrf_token_form=<?php echo attr_url(CsrfUtils::collectCsrfToken()); ?>'>
+			<input type="hidden" name="csrf_token_form"
+				value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+			<p>
+            <?php
+            $type = '';
+            $id = '';
+            if ($patient) {
+                $id = $patient;
+                $type = 'patient';
+            } elseif ($encounterid) {
+                $id = $encounterid;
+                $type = 'encounter';
+            } elseif ($formid) {
+                $id = $formid;
+                $type = 'form';
+            } elseif ($issue) {
+                $id = $issue;
+                $type = 'issue';
+            } elseif ($document) {
+                $id = $document;
+                $type = 'document';
+            } elseif ($payment) {
+                $id = $payment;
+                $type = 'payment';
+            } elseif ($billing) {
+                $id = $billing;
+                $type = 'invoice';
+            } elseif ($transaction) {
+                $id = $transaction;
+                $type = 'transaction';
+            }
+
+            $ids = explode(",", $id);
+            if (count($ids) > 1) {
+                $type .= 's';
+            }
+
+            $msg = 'You have selected ' . count($ids) . ' ' . $type . " to delete. Are you sure you want to continue?";
+            echo xlt($msg);
+            ?>
             </p>
-            <div class="btn-group">
-                <a href="#" onclick="submit_form()" class="btn btn-sm btn-delete btn-danger"><?php echo xlt('Yes, Delete and Log'); ?></a>
-                <a href='#' class="btn btn-sm btn-secondary btn-cancel" onclick="popup_close();"><?php echo xlt('No, Cancel');?></a>
+			<div class="btn-group">
+                <button onclick="submit_form()" class="btn btn-sm btn-primary mr-2"><?php echo xlt('Yes'); ?></button>
+                <button class="btn btn-sm btn-secondary" onclick="popup_close();"><?php echo xlt('No');?></button>
             </div>
```

Comment:
Really should give buttons inside a form a button type even with an onclick event that may prevent default. Default button type in a form is submit.
Also I believe the popup is a dialog and while you're here maybe you could convert the dialog to return a promise to deal with clean up instead of the javascript alert().
Dialog can return a promise on show or close.

Reply:
> Really should give buttons inside a form a button type even with an onclick event that may prevent default. Default button type in a form is submit.

OK

> Also I believe the popup is a dialog and while you're here maybe you could convert the dialog to return a promise to deal with clean up instead of the javascript alert().
> Dialog can return a promise on show or close.

That seems out of scope for the purpose of this PR - I prefer not to overload PRs. Also, I'm not clear on what you mean here.

Reply:
that's okay(I took a shot). I was looking to get rid of the confirm alert. no worries, i'll get to it one day:)